src/ast.hpp

```diff
@@ -542,221 +630,299 @@ namespace Sass {
              keyword_.compare("@-o-keyframes") == 0 ||
              keyword_.compare("@keyframes") == 0;
     }
+    ATTACH_AST_OPERATIONS(Directive)
     ATTACH_OPERATIONS()
   };
 
   ///////////////////////////////////////////////////////////////////////
   // Keyframe-rules -- the child blocks of "@keyframes" nodes.
   ///////////////////////////////////////////////////////////////////////
-  class Keyframe_Rule : public Has_Block {
-    ADD_PROPERTY(Selector*, selector)
+  class Keyframe_Rule_Ref : public Has_Block_Ref {
+    ADD_PROPERTY(Selector_Obj, selector2)
```

Comment:
Used during dev since I had to replace each class with an updated one without breaking stuff. Left for now (will rename sortly before PR finally ready). Used to differentiate regular `selector` from `Keyframe_Rule`. Easier to track where variables are passed and used.